lib/segment/src/common/rocksdb_buffered_update_wrapper.rs

```diff
@@ -77,4 +77,40 @@ impl DatabaseColumnScheduledUpdateWrapper {
     pub fn lock_db(&self) -> LockedDatabaseColumnWrapper {
         self.db.lock_db()
     }
+
+    pub fn get<K>(&self, key: K) -> OperationResult<Vec<u8>>
+    where
+        K: AsRef<[u8]>,
+    {
+        if let Some(value) = self.insert_pending_persistence.lock().get(key.as_ref()) {
+            return Ok(value.clone());
+        }
+        if self
+            .deleted_pending_persistence
+            .lock()
```

Comment:
It looks like we basically always lock these two mutexes together.

How about using a single mutex for both pending insertions and deletions?

Reply:
I think a single Mutex is indeed better perf. wise :+1: 

I will do it in a follow up request to get a cleaner diff.

Reply:
https://github.com/qdrant/qdrant/pull/5915